packages/kit/types/ambient.d.ts

```diff
@@ -263,6 +265,17 @@ declare module '$app/navigation' {
 	 * `afterNavigate` must be called during a component initialization. It remains active as long as the component is mounted.
 	 */
 	export function afterNavigate(callback: (navigation: AfterNavigate) => void): void;
+
+	/**
+	 * A lifecycle function that runs the supplied `callback` immediately before we navigate to a new URL.
+	 *
+	 * If a function is returned from the callback, it will be called once the DOM has updated.
```

Comment:
We should document the promise behavior (and also the various footguns, though is this the right place?)

```suggestion
	 * A lifecycle function that runs the supplied `callback` immediately before we navigate to a new URL.
	 *
	 * If you return a Promise, SvelteKit will wait for it to resolve before completing the navigation.
	 *
	 * If a function is returned from the callback, it will be called once the DOM has updated.
```